components/wso2is7.key.manager/src/main/java/org/wso2/is7/client/WSO2IS7ConnectorConfiguration.java

```diff
@@ -75,6 +76,69 @@ public List<ConfigurationDto> getConnectionConfigurations() {
+        List<ConfigurationDto> basicAuthValues = new ArrayList<>();
+        basicAuthValues.add(new ConfigurationDto("Username", "Username", "input",
+                "Username of admin user", "", true, false, Collections.emptyList(),
+                false));
+        basicAuthValues.add(new ConfigurationDto("Password", "Password", "input",
+                "Password of Admin user", "", true, true, Collections.emptyList(),
+                false));
+        ConfigurationDto basicAuthConfigurationDto = new ConfigurationDto("BasicAuth",
+                "Basic Authentication", "labelOnly", "Select to use basic authentication",
+                "", false, false, basicAuthValues, true);
+
+        List<ConfigurationDto> certBasedAuthValues = new ArrayList<>();
+        certBasedAuthValues.add(new ConfigurationDto("ServerWide", "Server-Wide Certificate",
+                "labelOnly", "Uses the configured global server certificate", "",
+                false, false, Collections.emptyList(), false));
+
+        certBasedAuthValues.add(new ConfigurationDto("TenantWide", "Tenant-Wide Certificate",
+                "labelOnly", "Tenant wide certificate for mutual TLS authentication", "",
+                false, false, Collections.singletonList(
+                        new ConfigurationDto("TenantWide", "Add a new certificate",
+                                "certificate", "Tenant wide certificate for mutual TLS authentication",
+                                "", true, false, Collections.emptyList(), false)),
+                false));
+        ConfigurationDto certificateBasedAuthConfigurationDto = new ConfigurationDto("Mutual-TLS",
+                "MTLS Authentication", "labelOnly", "Select to use MTLS authentication",
+                "", false, false, Arrays.asList(
+                        new ConfigurationDto("IdentityUser", "Identity Username", "input",
+                                "Username of identity user who belongs to the same tenant domain",
+                                "", true, false, Collections.emptyList(), false),
+                        new ConfigurationDto("Mutual-TLS",
+                                "Select a Certificate Type", "options",
+                                "", "", true, false,
+                                certBasedAuthValues, true)),
+                false);
+
+        List<ConfigurationDto> authValues = new ArrayList<>();
+        authValues.add(basicAuthConfigurationDto);
+        authValues.add(certificateBasedAuthConfigurationDto);
+
+        configurationDtoList.add(new ConfigurationDto("Authentication",
+                "Authentication Type", "dropdown", "Select the authentication type",
+                "BasicAuth", true, false, authValues, true));
+        configurationDtoList.add(new ConfigurationDto("api_resource_management_endpoint",
+                "WSO2 Identity Server 7 API Resource Management Endpoint", "input",
```

Comment:
cant we move these into the old connector configurations and remove existing basic auth configurations from connector configurations?

Reply:
as discussed serving both types of configurations will require significant changes to km connector UI. hence lets try renaming the method to something like "getAdvancedConfigurations" and rename the REST API field names as well.

Reply:
fixed